ethers-providers/src/rpc/transports/common.rs

```diff
@@ -144,51 +144,54 @@ impl<'de: 'a, 'a> Deserialize<'de> for Response<'a> {
                     match key {
                         "jsonrpc" => {
                             if jsonrpc {
-                                return Err(de::Error::duplicate_field("jsonrpc"))
+                                return Err(de::Error::duplicate_field("jsonrpc"));
                             }
 
                             let value = map.next_value()?;
                             if value != "2.0" {
-                                return Err(de::Error::invalid_value(Unexpected::Str(value), &"2.0"))
+                                return Err(de::Error::invalid_value(
+                                    Unexpected::Str(value),
+                                    &"2.0",
+                                ));
                             }
 
                             jsonrpc = true;
                         }
                         "id" => {
                             if id.is_some() {
-                                return Err(de::Error::duplicate_field("id"))
+                                return Err(de::Error::duplicate_field("id"));
                             }
 
                             let value: u64 = map.next_value()?;
                             id = Some(value);
                         }
                         "result" => {
                             if result.is_some() {
-                                return Err(de::Error::duplicate_field("result"))
+                                return Err(de::Error::duplicate_field("result"));
                             }
 
                             let value: &RawValue = map.next_value()?;
                             result = Some(value);
                         }
                         "error" => {
                             if error.is_some() {
-                                return Err(de::Error::duplicate_field("error"))
+                                return Err(de::Error::duplicate_field("error"));
                             }
 
-                            let value: JsonRpcError = map.next_value()?;
-                            error = Some(value);
+                            let value: Option<JsonRpcError> = map.next_value()?;
```

Comment:
this is the part that matters in this file. the rest is just cargo fmt running whenever my editor saves